### PR TITLE
approval state of merge requests

### PR DESCRIPTION
**Pull request by granular-ryanbonham**
Adds Support for getting the approval state of merge requests

https://docs.gitlab.com/ee/api/merge_request_approvals.html#get-the-approval-state-of-merge-requests

### CODE REVIEW

File: merge_request_approvals.go [hunk 1]
```diff
@@ -76,6 +76,30 @@ type MergeRequestApprovalRule struct {
...
+	RuleType             string               `json:"rule_type"`
+	EligibleApprovers    []*BasicUser         `json:"eligible_approvers"`
+	ApprovalsRequired    int                  `json:"approvals_required"`
+	SourceRule           *ProjectApprovalRule `json:"source_rule"`
+	Users                []*BasicUser         `json:"users"`
+	Groups               []*Group             `json:"groups"`
+	ContainsHiddenGroups bool                 `json:"contains_hidden_groups"`
+	ApprovedBy           []*BasicUser         `json:"approved_by"`
+	Approved             bool                 `json:"approved"`
+}
```
Comment (svanharmelen) [inline merge_request_approvals.go]: Let's not create a new struct, but instead add the missing fields to the `MergeRequestApprovalRule` struct and reuse that one here.

File: merge_request_approvals.go [hunk 2]
```diff
@@ -236,6 +260,31 @@ func (s *MergeRequestApprovalsService) GetApprovalRules(pid interface{}, mergeRe
...
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/merge_requests/%d/approval_state", pathEscape(project), mergeRequest)
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var par *MergeRequestApprovalState
```
Comment (svanharmelen) [inline merge_request_approvals.go]: Maybe call this `pas`, for this is about the state, not a rule?

```suggestion
	var pas *MergeRequestApprovalState
```

File: merge_request_approvals.go (hunk 1 above)
Comment (granular-ryanbonham) [reply merge_request_approvals.go]: Done

File: merge_request_approvals.go (hunk 2 above)
Comment (granular-ryanbonham) [reply merge_request_approvals.go]: Done

**Comment by granular-ryanbonham** [comment]
I addressed both comments. Let me know if you want any other changes. Thanks

**Comment by sluongng** [comment]
Thanks for doing this. I made an ugly vendor patch downstream but didnt have time to merge back

**Pull request merged**